library/ssl_msg.c

```diff
@@ -1888,7 +1888,7 @@ int mbedtls_ssl_fetch_input( mbedtls_ssl_context *ssl, size_t nb_want )
         {
             len = in_buf_len - ( ssl->in_hdr - ssl->in_buf );
 
-            if( mbedtls_ssl_is_handshake_over( ssl ) == 0 )
+            if( ssl->state != MBEDTLS_SSL_HANDSHAKE_OVER  )
```

Comment:
I would rather keep `mbedtls_ssl_is_handshake_over()` by default and change to `ssl->state !=/== MBEDTLS_SSL_HANDSHAKE_OVER` only where necessary for the writing and reading of post-handshake messages. Thus in `mbedtls_ssl_handshake()`, `mbedtls_ssl_handshake_step()`, `mbedtls_ssl_read/write()` and this should be about it (but I may miss other here).